src/dataprod_config.jl

```diff
@@ -279,83 +291,133 @@ function analysis_runs end
+"""
+cal_groupings_default(data::LegendData) = _groupings_runs(data, :cal)
+export cal_groupings_default
+
+
+const MaybeFileKey = Union{FileKey, Missing}
 const _cached_runinfo = LRU{UInt, Table}(maxsize = 300)
 
 """
     runinfo(data::LegendData)::Table
     runinfo(data::LegendData, runsel::RunSelLike)::NamedTuple
     runinfo(data::LegendData, filekey::FileKey)::NamedTuple
 
-Get the run information for `data`.
+Get the run information for `data` based on various selection criteria.
+
+# Arguments
+- `data::LegendData`: The dataset to query run information from.
+- `runlist::Symbol = :valid`: Specifies which runlist from `data.metadata.datasets.runlists` to use for filtering. Defaults to `:valid`.
+
+# Returns
+A table of run information with one named tuple per category (e.g. `:cal`, `:phy`), each containing `startkey`, `livetime`, and `is_analysis_run`
+
+# Example
+runinfo(data)                                   # full table of valid runs
+runinfo(data, :p03)                             # all runs in period p03
+runinfo(data, (:p03, :r005))                    # single-row Table for that run
+runinfo(data, (:p03, :r005, :phy))              # only the :phy entry
+runinfo(data, fk::FileKey)                      # same as above via FileKey
+runinfo(data; runlist = :custom_list)           # use a custom runlist
 """
-function runinfo(data::LegendData)
+function runinfo(data::LegendData; runlist::Symbol=:valid) #dataset
     get!(_cached_runinfo, objectid(data)) do
         # load runinfo
         rinfo = PropDict(data.metadata.datasets.runinfo)
-        parts_default = cal_groupings_default(data)
         nttype = @NamedTuple{startkey::MaybeFileKey, livetime::typeof(1.0u"s"), is_analysis_run::Bool}
+        phy_groupings = phy_groupings_default(data)
+        cal_groupings = cal_groupings_default(data)
+        
+        runlist_data = get(data.metadata.datasets.runlists, runlist, PropDict())
+        allowed = Set{Tuple{Symbol, Symbol, DataRun}}()
+        for (cat, periods) in runlist_data
+            for (p, rlist) in periods
+                for r in parse_runs(rlist)
+                    push!(allowed, (Symbol(cat), p, r))
+                end
+            end
+        end
+
         function make_row(p, r, ri)
             period::DataPeriod = DataPeriod(p)
             run::DataRun = DataRun(r)
             function get_cat_entry(cat)
                 if haskey(ri, cat)
                     fk = ifelse(haskey(ri[cat], :start_key), FileKey(data.name, period, run, cat, Timestamp(get(ri[cat], :start_key, 1))), missing)
-                    is_ana_run::Bool = (; period, run) in _analysis_runs(data, cat) && !ismissing(fk)
+                    is_ana_run = if cat == :phy
+                        (; period, run) in phy_groupings && !ismissing(fk)
+                    elseif cat == :cal 
+                        (; period, run) in cal_groupings && !ismissing(fk)
+                    else
+                        true
+                    end
```

Comment:
Does this only apply to `phy` and `cal`? If not, this could probably be combined to something like
```suggestion
                    is_ana_run::Bool = (; period, run) in _groupings_runs(data, cat) && !ismissing(fk)
```

Reply:
How important is speed here? Because currently i am only callingt` _groupings_runs` once, with your suggestion, each time `make_row` is called